Automattic/jetpack
# OG Tags: Fix notice when author object is not retrieved + PHPCS

Fixes notice reported:
```E_NOTICE: /home/public_html/wp-content/plugins/jetpack/functions.opengraph.php:308 - Trying to get property 'user_email' of non-object```
and prevents it in other places.

PHPCS for the rest of the file.

#### Changes proposed in this Pull Request:
* Verify the WP_User object is returned.

#### Testing instructions:
* With OG tags enabled (e.g. Sharing enbaled), visit an author profile page
* Check the OG tags - are they present?

#### Proposed changelog entry for your changes:
* OG Tags: Ensure the author's information is valid on author archive pages.
